source/kraken/tests/realtime_test.cpp

```diff
@@ -2740,3 +2740,380 @@ BOOST_AUTO_TEST_CASE(should_get_correct_base_stop_time_with_lollipop_II) {
     BOOST_CHECK_EQUAL(stoptime_rltm_B_bis.get_base_stop_time(), &vj->stop_time_list[3]);
     BOOST_CHECK_EQUAL(stoptime_rltm_D.get_base_stop_time(), &vj->stop_time_list[4]);
 }
+
+BOOST_AUTO_TEST_CASE(add_new_trip) {
+
+    using year = navitia::type::ValidityPattern::year_bitset;
+
+    ed::builder b("20190101");
+    auto& pt_data = *b.data->pt_data;
+
+    b.sa("A", 0, 0, true, true);
+    b.sa("B", 0, 0, true, true);
+    b.sa("C", 0, 0, true, true);
+    b.sa("D", 0, 0, true, true);
+    b.sa("E", 0, 0, true, true);
+    b.sa("F", 0, 0, true, true);
+    b.sa("G", 0, 0, true, true);
+    b.sa("H", 0, 0, true, true);
+    b.sa("I", 0, 0, true, true);
+    b.sa("J", 0, 0, true, true);
+
+    b.vj("1").uri("vj:1")
+                 ("stop_point:A", "08:00"_t)
+                 ("stop_point:B", "08:30"_t)
+                 ("stop_point:C", "09:00"_t)
+                 ("stop_point:D", "09:30"_t);
+
+    // Add company
+    std::string comp_name = "comp_name";
+    std::string comp_uri = "comp_uri";
+    navitia::type::Company* cmp1 = new navitia::type::Company();
+    cmp1->line_list.push_back(b.lines["1"]);
+    cmp1->idx = b.data->pt_data->companies.size();
+    cmp1->name = comp_name;
+    cmp1->uri = comp_uri;
+    b.data->pt_data->companies.push_back(cmp1);
+    b.data->pt_data->companies_map[comp_uri] = cmp1;
+    b.lines["1"]->company_list.push_back(cmp1);
+
+    // Add physical mode
+    std::string phy_mode_name = "physical_mode_name";
+    std::string phy_mode_uri = "physical_mode_uri";
+    navitia::type::PhysicalMode* phy_mode = new navitia::type::PhysicalMode();
+    phy_mode->idx = b.data->pt_data->physical_modes.size();
+    phy_mode->name = phy_mode_name;
+    phy_mode->uri = phy_mode_uri;
+    phy_mode->vehicle_journey_list.push_back(b.data->pt_data->vehicle_journeys[0]);
+    b.data->pt_data->physical_modes.push_back(phy_mode);
+    b.data->pt_data->physical_modes_map[phy_mode_uri] = phy_mode;
+    b.lines["1"]->physical_mode_list.push_back(phy_mode);
+
+    b.make();
+
+    navitia::routing::RAPTOR raptor(*(b.data));
+    ng::StreetNetwork sn_worker(*b.data->geo_ref);
+
+    auto compute = [&](const char* datetime, const std::string& from, const std::string& to) {
```

Comment:
add a std::string if you prefer. This case is used in all realtime_test.cpp file.
We can do even better, like extract the _compute(...)_ function so that use in all code. But not in this PR, it is already heavy changes.

Reply:
:+1: changed to string
Mutualizing the method in test_utils to come in a JIRA ticket on test refactor

Reply:
JIRA : https://jira.kisio.org/browse/NAVP-1228